app/controllers/review_mapping_controller.rb

```diff
@@ -455,16 +482,36 @@ def start_self_review
   private
 
   def assign_reviewers_for_team(assignment_id, review_strategy, participants_hash)
-    if ReviewResponseMap.where(reviewed_object_id: assignment_id, calibrate_to: 0)
-                        .where('created_at > :time',
-                               time: @@time_create_last_review_mapping_record).size < review_strategy.reviews_needed
-
+    if check_eligibility(assignment_id,review_strategy)
       participants_with_insufficient_review_num = []
       participants_hash.each do |participant_id, review_num|
         participants_with_insufficient_review_num << participant_id if review_num < review_strategy.reviews_per_student
       end
-      unsorted_teams_hash = {}
+      #can be added in a method
+      teams_hash = generate_teams_hash(assignment_id)
+      #can be added in a method
+      teams_hash = teams_hash_modified(participants_with_insufficient_review_num, teams_hash, assignment_id)
+    end
+    @@time_create_last_review_mapping_record = last_created_time_review_mapping_record(assignment_id)
+  end
+
+  # This method is used to check eligibility 
+  def check_eligibility(assignment_id,review_strategy)
```

Comment:
What eligibility does this method check? Can you rename it to more informative one?